### PR TITLE
fix: add timezone to timestamps

### PR DESCRIPTION
**Pull request by rasendubi**
---
labels: mergeable
---
[//]: #  (Link to the issue corresponding to this chunk of work)
Fixes: #__issue__

## Motivation and Context
[//]: #  (Why is this change required? What problem does it solve?)

This fixes timestamps, so they get properly serialized with timezone information.

## Description
[//]: # (Describe your changes in detail)

```
>>> datetime.datetime.now(datetime.timezone.utc).isoformat()
'2024-08-20T17:58:45.374349+00:00'
>>> datetime.datetime.utcnow().isoformat()
'2024-08-20T18:01:31.482134'
```

## How has this been tested?
[//]: # (Please describe in detail how you tested your changes)
Not tested.

[//]: # (OPTIONAL)
[//]: # (Add one or multiple labels: enhancement, refactoring, bugfix)

### CODE REVIEW

File: eppo_client/client.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def get_assignment_detail(
             "featureFlag": flag_key,
             "variation": result.variation.key if result and result.variation else None,
             "subject": subject_key,
-            "timestamp": datetime.datetime.utcnow().isoformat(),
+            "timestamp": datetime.datetime.now(datetime.timezone.utc).isoformat(),
```
Comment (leoromanovsky) [inline eppo_client/client.py:225]: A desirable test would be to refactor this into a function and the unit test would have consistent time:

```
import unittest
import datetime
from freezegun import freeze_time

def get_utc_timestamp():
    return datetime.datetime.now(datetime.timezone.utc).isoformat()

class TestUTCTimestamp(unittest.TestCase):
    @freeze_time("2023-08-20 14:30:45.123456+00:00")
    def test_get_utc_timestamp(self):
        expected = "2023-08-20T14:30:45.123456+00:00"
        result = get_utc_timestamp()
        self.assertEqual(result, expected)

    def test_timezone_present(self):
        result = get_utc_timestamp()
        self.assertIn('+00:00', result)

    def test_iso_format(self):
        result = get_utc_timestamp()
        try:
            datetime.datetime.fromisoformat(result)
        except ValueError:
            self.fail("get_utc_timestamp() did not return a valid ISO format")
```

**Review by schmit — APPROVED** [review]
Great catch -- perhaps we should refactor into a `get_time()` function (probably with a better name) and call that so we don't risk calling the wrong function in the future (although of course we could forget that the `get_time` function exists...

Also don't forget a version bump!

**Comment by rasendubi** [comment]
@schmit we actually have a good `utcnow()` in `eval.py`.

I'm signing off for today so didn't have much time for a nice fix with tests. I can improve it tomorrow or feel free to pick it up

File: eppo_client/client.py (hunk 1 above)
Comment (rasendubi) [reply eppo_client/client.py]: I added two higher-level tests asserting that event timestamps carry the timezone information (ultimately, we don't care whether string uses `+00:00` or `Z` suffix):
```python
event = mock_logger.log_assignment.call_args.args[0]
timestamp = datetime.datetime.fromisoformat(event["timestamp"])
assert timestamp.tzinfo == datetime.timezone.utc
```

**Comment by rasendubi** [comment]
@schmit @leoromanovsky refactored this into a function + added tests

**Review by leoromanovsky — APPROVED** [review]
👍 👍 👍

**Pull request merged**